src/software/multithreading/observer_test.cpp

```diff
@@ -46,3 +46,33 @@ TEST(Observer, receiveValue_value_not_yet_available)
     ASSERT_TRUE(result);
     EXPECT_EQ(202, *result);
 }
+
+TEST(Observer, getDataReceivedPerSecond_time_buffer_filled)
+{
+    TestObserver test_observer;
+    for (unsigned int i = 0; i < TestObserver::TIME_BUFFER_SIZE; i++)
+    {
+        test_observer.receiveValue(i);
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    }
+    std::cout << test_observer.getDataReceivedPerSecond() << std::endl;
+    std::cout << test_observer.getDataReceivedPerSecond() << std::endl;
+    EXPECT_NEAR(test_observer.getDataReceivedPerSecond(), 5 / (4 * 0.01), 10);
+}
+
+TEST(Observer, getDataReceivedPerSecond_time_buffer_empty)
+{
+    TestObserver test_observer;
+    EXPECT_EQ(test_observer.getDataReceivedPerSecond(), 0);
+}
+
+TEST(Observer, getDataReceivedPerSecond_time_buffer_partially_empty)
+{
+    TestObserver test_observer;
+    for (unsigned int i = 0; i < TestObserver::TIME_BUFFER_SIZE / 2; i++)
+    {
+        test_observer.receiveValue(i);
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    }
+    EXPECT_EQ(test_observer.getDataReceivedPerSecond(), 0);
```

Comment:
I don't understand why this would return `0`? Seems like it should return a value around `100` since it received a piece of data every 10ms = 100 times per second. This calculation should be possible as long as the buffer has at least 2 values (which it would in this case).

Reply:
oops